### PR TITLE
OI4-55

### PR DESCRIPTION
**Pull request by vdedominicis-EH**

### CODE REVIEW

File: packages/oi4-oec-service-node/src/Utilities/Helpers/ClientPayloadHelper.ts [hunk 1]
```diff
@@ -63,34 +63,25 @@ export class ClientPayloadHelper {
         return {abortSending: false, payload: payload};
     }
 
-    createLicenseSendResourcePayload(applicationResources: IOI4ApplicationResources, filter: string, dataSetWriterIdFilter: number, resource: string): ValidatedPayload {
+    createLicenseSendResourcePayload(applicationResources: IOI4ApplicationResources, dataSetWriterIdFilter: number, resource: string, filter: string = undefined): ValidatedPayload {
         const payload: IOPCUAPayload[] = [];
+        let licenses: ILicenseObject[] = applicationResources.license;
 
         if (Number.isNaN(dataSetWriterIdFilter)) { // Try to filter with licenseId
```
Comment (Weinschenk) [inline packages/oi4-oec-service-node/src/Utilities/Helpers/ClientPayloadHelper.ts]: Check for NaN is obsolete as it is always a number (it used to be string in the past)

File: packages/oi4-oec-service-node/src/Utilities/Helpers/ClientPayloadHelper.ts [hunk 2]
```diff
@@ -63,34 +63,25 @@ export class ClientPayloadHelper {
         return {abortSending: false, payload: payload};
     }
 
-    createLicenseSendResourcePayload(applicationResources: IOI4ApplicationResources, filter: string, dataSetWriterIdFilter: number, resource: string): ValidatedPayload {
+    createLicenseSendResourcePayload(applicationResources: IOI4ApplicationResources, dataSetWriterIdFilter: number, resource: string, filter: string = undefined): ValidatedPayload {
```
Comment (Weinschenk) [inline packages/oi4-oec-service-node/src/Utilities/Helpers/ClientPayloadHelper.ts]: According to the spec, we need to "filters" for the license; <oi4Identifier> and <licenseId>.
It is possible to only provide the oi4Identifier, but not the licenseId. Therefore I suggest the following method signature:
```suggestion
    createLicenseSendResourcePayload(applicationResources: IOI4ApplicationResources, oi4Identifier?: string, licenseId?: string = undefined): ValidatedPayload {
```

File: packages/oi4-oec-service-node/src/Utilities/Helpers/ClientPayloadHelper.ts (hunk 1 above)
Comment (Weinschenk) [inline packages/oi4-oec-service-node/src/Utilities/Helpers/ClientPayloadHelper.ts]: Please rework the nested if statements.

**Review by Weinschenk — CHANGES_REQUESTED** [review]

File: packages/oi4-oec-service-node/src/Utilities/Helpers/ClientPayloadHelper.ts (hunk 1 above)
Comment (vdedominicis-EH) [reply packages/oi4-oec-service-node/src/Utilities/Helpers/ClientPayloadHelper.ts]: Fixed

File: packages/oi4-oec-service-node/src/Utilities/Helpers/ClientPayloadHelper.ts (hunk 2 above)
Comment (vdedominicis-EH) [reply packages/oi4-oec-service-node/src/Utilities/Helpers/ClientPayloadHelper.ts]: discussed and fixed together

File: packages/oi4-oec-service-node/src/Utilities/Helpers/ClientPayloadHelper.ts (hunk 1 above)
Comment (vdedominicis-EH) [reply packages/oi4-oec-service-node/src/Utilities/Helpers/ClientPayloadHelper.ts]: done implicitly while dealing with the above point

**Review by Weinschenk — APPROVED** [review]

**Pull request merged**